### PR TITLE
W1-phongle-Create login screen and sign up screen

### PR DESCRIPTION
**Pull request by at-phongle**
#### Related issues (Optional)
Task https://github.com/akaj-team/intern-17k3/issues/1

#### Description (Required)
- [x] Create Login screen
- [x] Create Sign up screen

### CODE REVIEW

File: app/src/at-phongle/java/vn/asiantech/internship/MainActivity.java [hunk 1]
```diff
@@ -1,14 +1,21 @@
 package vn.asiantech.internship;
 
-import android.app.Activity;
+import android.content.Intent;
 import android.os.Bundle;
+import android.support.v4.app.FragmentManager;
```
Comment (at-dev-support-01) [inline app/src/at-phongle/java/vn/asiantech/internship/MainActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.support.v4.app.FragmentManager'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-phongle/res/drawable/ic_launcher_foreground.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:aapt="http://schemas.android.com/aapt"
+    android:width="108dp"
+    android:height="108dp"
+    android:viewportHeight="108"
+    android:viewportWidth="108">
+    <path
+        android:fillType="evenOdd"
```
Comment (at-dev-support-01) [inline app/src/at-phongle/res/drawable/ic_launcher_foreground.xml]: WARNING: [UnusedAttribute] Attribute `fillType` is only used in API level 24 and higher (current min is 15)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-phongle/res/drawable/ic_launcher_foreground.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        android:pathData="M32,64C32,64 38.39,52.99 44.13,50.95C51.37,48.37 70.14,49.57 70.14,49.57L108.26,87.69L108,109.01L75.97,107.97L32,64Z"
+        android:strokeColor="#00000000"
+        android:strokeWidth="1">
+        <aapt:attr name="android:fillColor">
+            <gradient
+                android:endX="78.5885"
```
Comment (at-dev-support-01) [inline app/src/at-phongle/res/drawable/ic_launcher_foreground.xml]: WARNING: [UnusedAttribute] Attribute `endX` is only used in API level 24 and higher (current min is 15)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-phongle/res/drawable/ic_launcher_foreground.xml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                android:endY="90.9159"
```
Comment (at-dev-support-01) [inline app/src/at-phongle/res/drawable/ic_launcher_foreground.xml]: WARNING: [UnusedAttribute] Attribute `endY` is only used in API level 24 and higher (current min is 15)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-phongle/res/drawable/ic_launcher_foreground.xml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+                android:startX="48.7653"
```
Comment (at-dev-support-01) [inline app/src/at-phongle/res/drawable/ic_launcher_foreground.xml]: WARNING: [UnusedAttribute] Attribute `startX` is only used in API level 24 and higher (current min is 15)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-phongle/res/drawable/ic_launcher_foreground.xml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+                android:startY="61.0927"
```
Comment (at-dev-support-01) [inline app/src/at-phongle/res/drawable/ic_launcher_foreground.xml]: WARNING: [UnusedAttribute] Attribute `startY` is only used in API level 24 and higher (current min is 15)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-phongle/res/drawable/ic_launcher_foreground.xml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+                android:type="linear">
+                <item
+                    android:color="#44000000"
+                    android:offset="0.0" />
```
Comment (at-dev-support-01) [inline app/src/at-phongle/res/drawable/ic_launcher_foreground.xml]: WARNING: [UnusedAttribute] Attribute `offset` is only used in API level 24 and higher (current min is 15)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-phongle/res/drawable/ic_launcher_foreground.xml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+                <item
+                    android:color="#00000000"
+                    android:offset="1.0" />
```
Comment (at-dev-support-01) [inline app/src/at-phongle/res/drawable/ic_launcher_foreground.xml]: WARNING: [UnusedAttribute] Attribute `offset` is only used in API level 24 and higher (current min is 15)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-phongle/res/drawable/ic_launcher_foreground.xml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+            </gradient>
+        </aapt:attr>
+    </path>
+    <path
+        android:fillColor="#FFFFFF"
+        android:fillType="nonZero"
```
Comment (at-dev-support-01) [inline app/src/at-phongle/res/drawable/ic_launcher_foreground.xml]: WARNING: [UnusedAttribute] Attribute `fillType` is only used in API level 24 and higher (current min is 15)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

**Comment by at-phongle** [comment]
@at-vinhhuynh Please review this task
Thanks

File: app/src/at-phongle/java/vn/asiantech/internship/login/LoginActivity.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+public class LoginActivity extends AppCompatActivity {
+    private ImageView mImgBack;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+        mImgBack = findViewById(R.id.imgBack);
+        addFragment(LoginFragment.getInstance(), false);
+
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/java/vn/asiantech/internship/login/LoginActivity.java]: please remove this line

File: app/src/at-phongle/java/vn/asiantech/internship/login/LoginFragment.java [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    public static Fragment getInstance() {
+        return new LoginFragment();
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        super.onCreateView(inflater, container, savedInstanceState);
+        View view = inflater.inflate(R.layout.fragment_login, container, false);
+
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/java/vn/asiantech/internship/login/LoginFragment.java]: Please remove this break line

File: app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import vn.asiantech.internship.R;
+
+public class SingUpFragment extends Fragment implements View.OnClickListener, TextWatcher {
+    private CheckBox mChkTermService;
+    private ImageView mImgNext;
+    private EditText mEdtPhone, mEdtEmail, mEdtFullName;
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java]: Please separate this line to 3 lines

File: app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        super.onCreateView(inflater, container, savedInstanceState);
+        View v = inflater.inflate(R.layout.fragment_signup, container, false);
+        mChkTermService = v.findViewById(R.id.chkTermServices);
+        mChkTermService.setOnClickListener(this);
+        mImgNext = v.findViewById(R.id.imgNext);
+        mEdtEmail = v.findViewById(R.id.edtEmail);
+        mEdtFullName = v.findViewById(R.id.edtFullName);
+        mEdtPhone = v.findViewById(R.id.edtPhone);
+        //handle edit text listener
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java]: Change comment to `// Handle edit text listener `

File: app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+        tvTermService.setText(Html.fromHtml(text), TextView.BufferType.SPANNABLE);
+        return v;
+    }
+
+    @Override
+    public void onClick(View v) {
+        mImgNext.setSelected(checkSignUp(mEdtPhone, mEdtFullName, mEdtEmail, mChkTermService));
+    }
+
+    protected boolean checkSignUp(TextView tv1, TextView tv2, TextView tv3, CheckBox chk) {
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java]: Please change `protected` to `private`

File: app/src/at-phongle/res/drawable/bg_bt_login_facebook.xml [hunk 15]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/res/drawable/bg_bt_login_facebook.xml]: Please change file name to `bg_btn_login_facebook.xml`

File: app/src/at-phongle/res/drawable/bg_bt_login_phone.xml [hunk 16]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <solid android:color="@color/blue_facebook" />
+    <stroke android:width="0.5dp" />
+    <corners android:radius="5dp" />
+
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/res/drawable/bg_bt_login_phone.xml]: remove this line

File: app/src/at-phongle/res/layout/activity_login.xml [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+            android:layout_below="@+id/imgLogo"
+            android:layout_centerHorizontal="true"
+            android:layout_marginTop="50dp"
+            android:text="@string/tv_banner"
+            android:textColor="@android:color/holo_blue_bright" />
+
+    </RelativeLayout>
+
+    <FrameLayout
+        android:id="@+id/frlContainer"
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/res/layout/activity_login.xml]: `frlContainer` to `frContainer`

File: app/src/at-phongle/res/layout/fragment_login.xml [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+        android:text="@string/btn_login_facebook"
+        android:textColor="@android:color/white"
+        android:textSize="17dp"
+        android:textStyle="bold" />
+
+    <RelativeLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_margin="10dp"
+        android:gravity="center">
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/res/layout/fragment_login.xml]: Please remove this attr `android:gravity="center"`

File: app/src/at-phongle/res/layout/fragment_signup.xml [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+                android:layout_height="25dp"
+                android:src="@drawable/ic_australia" />
+
+            <ImageView
+                android:layout_width="10dp"
+                android:layout_height="10dp"
+                android:layout_gravity="center"
+                android:layout_margin="5dp"
+                android:src="@drawable/ic_drop_down" />
+
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/res/layout/fragment_signup.xml]: remove this line

File: app/src/at-phongle/res/layout/fragment_signup.xml [hunk 20]
```diff
@@ -0,0 +1,97 @@
...
+        </LinearLayout>
+
+        <EditText
+            android:id="@+id/edtPhone"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:hint="@string/edt_phone"
+            android:inputType="phone"
+            android:textCursorDrawable="@drawable/cursor_edt_sing_up" />
+
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/res/layout/fragment_signup.xml]: remove this line

File: app/src/at-phongle/res/layout/fragment_signup.xml [hunk 21]
```diff
@@ -0,0 +1,97 @@
...
+            android:button="@drawable/selector_chk_term_service"
+            android:scaleX="1.5"
+            android:scaleY="1.5" />
+
+        <TextView
+            android:id="@+id/tvTermService"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:textColor="@android:color/black" />
+
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/res/layout/fragment_signup.xml]: remove this line

File: app/src/at-phongle/res/values/strings.xml [hunk 22]
```diff
@@ -1,3 +1,16 @@
...
+    <string name="tv_welcome">Welcome to urbi</string>
+    <string name="tv_sign_up">Sign up for an account</string>
+    <string name="tv_new_urbi">New to urbi?&#160;</string>
+    <string name="or">or</string>
+    <string name="edt_email">Email</string>
+    <string name="edt_phone"> +67 Enter mobile number</string>
+    <string name="tv_banner">a bike for everyone , always</string>
+    <string name="tv_term_services">I have read and agree with the terms and conditions</string>
 </resources>
+
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/res/values/strings.xml]: remove this line

File: app/src/main/res/values/strings.xml [hunk 23]
```diff
@@ -1,3 +1,2 @@
 <resources>
-
```
Comment (at-vinhhuynh) [inline app/src/main/res/values/strings.xml]: Please revert this file

**Comment by at-phongle** [comment]
@at-vinhhuynh  please review this 
Thanks

File: app/src/at-phongle/res/drawable/bg_btn_login_facebook.xml [hunk 24]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <solid android:color="#FAFAFA" />
+    <stroke
+        android:width="1dp"
+        android:color="#1E90FF" />
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/res/drawable/bg_btn_login_facebook.xml]: Please dèfine color on `color.xml`

**Comment by at-phongle** [comment]
@at-vinhhuynh  please review this, thanks you

File: app/src/at-phongle/java/vn/asiantech/internship/login/LoginActivity.java [hunk 25]
```diff
@@ -0,0 +1,49 @@
...
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+        mImgBack = findViewById(R.id.imgBack);
+        mImgBack.setOnClickListener(this);
+        replaceFragment(LoginFragment.getInstance(), false);
+    }
+
+    protected void replaceFragment(Fragment fragment, boolean addBackStack) {
+        mImgBack.setSelected(addBackStack);
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/java/vn/asiantech/internship/login/LoginActivity.java]: why add this line here?

File: app/src/at-phongle/res/layout/fragment_login.xml [hunk 26]
```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_centerInParent="true"
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/res/layout/fragment_login.xml]: Please remove this line `android:layout_centerInParent="true"`

File: app/src/at-phongle/res/layout/fragment_login.xml [hunk 27]
```diff
@@ -0,0 +1,83 @@
...
+        android:text="@string/tv_welcome"
+        android:textColor="@android:color/black"
+        android:textStyle="bold" />
+
+    <Button
+        android:id="@+id/btnLoginFacebook"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@drawable/bg_btn_login_facebook"
+        android:stateListAnimator="@null"
```
Comment (at-dev-support-01) [inline app/src/at-phongle/res/layout/fragment_login.xml]: WARNING: [UnusedAttribute] Attribute `stateListAnimator` is only used in API level 21 and higher (current min is 15)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-phongle/res/layout/fragment_login.xml [hunk 28]
```diff
@@ -0,0 +1,83 @@
...
+            android:text="@string/or" />
+
+    </RelativeLayout>
+
+    <Button
+        android:id="@+id/btnLoginPhone"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@drawable/bg_btn_login_phone"
+        android:stateListAnimator="@null"
```
Comment (at-dev-support-01) [inline app/src/at-phongle/res/layout/fragment_login.xml]: WARNING: [UnusedAttribute] Attribute `stateListAnimator` is only used in API level 21 and higher (current min is 15)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

**Comment by at-phongle** [comment]
@at-vinhhuynh  please review this. thanks

File: app/src/at-phongle/java/vn/asiantech/internship/login/LoginActivity.java [hunk 29]
```diff
@@ -0,0 +1,45 @@
+package vn.asiantech.internship.login;
+
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentTransaction;
+import android.support.v7.app.AppCompatActivity;
+import android.view.KeyEvent;
```
Comment (at-dev-support-01) [inline app/src/at-phongle/java/vn/asiantech/internship/login/LoginActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.view.KeyEvent'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

**Comment by at-phongle** [comment]
@at-vinhhuynh  please review this . thank you

File: app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java [hunk 30]
```diff
@@ -0,0 +1,77 @@
...
+    private EditText mEdtEmail;
+    private EditText mEdtFullName;
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        super.onCreateView(inflater, container, savedInstanceState);
+        View v = inflater.inflate(R.layout.fragment_signup, container, false);
+        mChkTermService = v.findViewById(R.id.chkTermServices);
+        mChkTermService.setOnClickListener(this);
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java]: Please move this line to line 41

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finished review this PR, please check it.
thank you!

**Comment by at-vinhhuynh** [comment]
@at-huongnguyen 
I finished review this PR, please check it again.
thank you!

File: app/src/at-phongle/java/vn/asiantech/internship/login/LoginActivity.java [hunk 31]
```diff
@@ -0,0 +1,44 @@
...
+        fragmentTransaction.commit();
+    }
+
+    protected void setVisibilityBackButton(int visibility) {
+        mImgBack.setVisibility(visibility);
+    }
+
+    @Override
+    public void onClick(View v) {
+        onBackPressed();
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/login/LoginActivity.java]: Check v.getId() == R.id.imgBack before  call onBackPressed()

File: app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java [hunk 32]
```diff
@@ -0,0 +1,77 @@
...
+        mChkTermService = v.findViewById(R.id.chkTermServices);
+        mImgNext = v.findViewById(R.id.imgNext);
+        mEdtEmail = v.findViewById(R.id.edtEmail);
+        mEdtFullName = v.findViewById(R.id.edtFullName);
+        mEdtPhone = v.findViewById(R.id.edtPhone);
+        // Handle edit text listener
+        mEdtEmail.addTextChangedListener(this);
+        mEdtPhone.addTextChangedListener(this);
+        mEdtFullName.addTextChangedListener(this);
+        mChkTermService.setOnClickListener(this);
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java]: Create initViews() and initListener() method

File: app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java [hunk 33]
```diff
@@ -0,0 +1,77 @@
...
+        // Set text with two color
+        TextView tvTermService = v.findViewById(R.id.tvTermService);
+        String text = "I have read and agree with the <font color='#00bfff'> <br/> term and conditions</font>. ";
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java]: Write to strings.xml file

File: app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java [hunk 34]
```diff
@@ -0,0 +1,77 @@
...
+        mImgNext.setSelected(checkSignUp(mEdtPhone, mEdtFullName, mEdtEmail, mChkTermService));
+    }
+
+    private boolean checkSignUp(TextView tv1, TextView tv2, TextView tv3, CheckBox chk) {
+        return (!(tv1.getText().toString().trim().isEmpty()) && !(tv2.getText().toString().trim().isEmpty()) && !(tv3.getText().toString().trim().isEmpty()) && chk.isChecked());
+    }
+
+    @Override
+    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+        mImgNext.setSelected(checkSignUp(mEdtPhone, mEdtFullName, mEdtEmail, mChkTermService));
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java]: Remove check in `beforeTextChanged ` method

File: app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java [hunk 35]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Override
+    public void onTextChanged(CharSequence s, int start, int before, int count) {
+        mImgNext.setSelected(checkSignUp(mEdtPhone, mEdtFullName, mEdtEmail, mChkTermService));
+    }
+
+    @Override
+    public void afterTextChanged(Editable s) {
+        mImgNext.setSelected(checkSignUp(mEdtPhone, mEdtFullName, mEdtEmail, mChkTermService));
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java]: Remove check in `afterTextChanged ` method

File: app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java [hunk 36]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        ((LoginActivity) getActivity()).setVisibilityBackButton(View.VISIBLE);
+    }
+
+    @Override
+    public void onClick(View v) {
+        mImgNext.setSelected(checkSignUp(mEdtPhone, mEdtFullName, mEdtEmail, mChkTermService));
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java]: Check id before call this

File: app/src/at-phongle/java/vn/asiantech/internship/login/LoginFragment.java [hunk 37]
```diff
@@ -0,0 +1,39 @@
...
+    @Override
+    public void onResume() {
+        super.onResume();
+        ((LoginActivity) getActivity()).setVisibilityBackButton(View.GONE);
+    }
+
+    @Override
+    public void onClick(View v) {
+        ((LoginActivity) getActivity()).replaceFragment(new SingUpFragment(), true);
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/login/LoginFragment.java]: Check id before call this

File: app/src/at-phongle/res/layout/fragment_signup.xml [hunk 38]
```diff
@@ -0,0 +1,103 @@
...
+        android:layout_height="wrap_content"
+        android:layout_marginTop="20dp"
+        android:background="@drawable/cursor_edt_sing_up"
+        android:hint="@string/edt_email"
+        android:inputType="textEmailAddress"
+        android:textCursorDrawable="@drawable/cursor_edt_sing_up" />
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="60dp"
```
Comment (at-huongnguyen) [inline app/src/at-phongle/res/layout/fragment_signup.xml]: should use wrap_content

File: app/src/at-phongle/res/layout/fragment_login.xml [hunk 39]
```diff
@@ -0,0 +1,84 @@
...
+        style="@style/Base.Widget.AppCompat.Button.Borderless"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@drawable/bg_btn_login_phone"
+        android:text="@string/btn_login_phone"
+        android:textColor="@android:color/holo_blue_dark"
+        android:textSize="17dp"
+        android:textStyle="bold" />
+
+    <LinearLayout
```
Comment (at-huongnguyen) [inline app/src/at-phongle/res/layout/fragment_login.xml]: Use spannable

File: app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java [hunk 40]
```diff
@@ -0,0 +1,92 @@
+package vn.asiantech.internship.login;
+
+import android.graphics.Color;
```
Comment (at-dev-support-01) [inline app/src/at-phongle/java/vn/asiantech/internship/login/SingUpFragment.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.graphics.Color'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

**Comment by at-phongle** [comment]
@at-huongnguyen  please review this. Thanks

**Pull request merged**